promql/promqltest/test.go

```diff
@@ -990,11 +990,22 @@ func (t *test) exec(tc testCommand, engine promql.QueryEngine) error {
 }
 
 func (t *test) execEval(cmd *evalCmd, engine promql.QueryEngine) error {
-	if cmd.isRange {
-		return t.execRangeEval(cmd, engine)
+	do := func() error {
+		if cmd.isRange {
+			return t.execRangeEval(cmd, engine)
+		}
+
+		return t.execInstantEval(cmd, engine)
+	}
+
+	if tt, ok := t.T.(*testing.T); ok {
```

Comment:
I didn't really follow this.  What are the circumstances where we have or don't have a `testing.T` ?